Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp

```diff
@@ -1043,17 +1043,28 @@ void fillVideoInfoColorimetryFromColorSpace(GstVideoInfo* info, const PlatformVi
 
 void configureVideoDecoderForHarnessing(const GRefPtr<GstElement>& element)
 {
-    auto elementHasProperty = [&](const char* name) -> bool {
-        return g_object_class_find_property(G_OBJECT_GET_CLASS(element.get()), name);
-    };
-
-    if (elementHasProperty("max-threads"))
+    if (gstElementHasProperty(element.get(), "max-threads"))
         g_object_set(element.get(), "max-threads", 1, nullptr);
 
-    if (elementHasProperty("max-errors"))
+    if (gstElementHasProperty(element.get(), "max-errors"))
         g_object_set(element.get(), "max-errors", 0, nullptr);
 }
 
+static bool gstObjectHasProperty(GstObject* gstObject, const char* name)
+{
+    return g_object_class_find_property(G_OBJECT_GET_CLASS(gstObject), name);
+}
+
+bool gstElementHasProperty(GstElement* element, const char* name)
+{
+    return gstObjectHasProperty(GST_OBJECT_CAST(element), name);
+}
+
+bool gstPadHasProperty(GstPad* pad, const char* name)
```

Comment:
Would this work if we create an overloaded `gstObjectHasPropety` taking also `GstPad` and `GstElement`?

Reply:
Hum, yes, good idea.